src/main/java/net/dries007/tfc/network/PacketAnvilUpdate.java

```diff
@@ -56,7 +57,8 @@ public void fromBytes(ByteBuf buffer)
         workProgress = buffer.readInt();
         workTarget = buffer.readInt();
         pos = BlockPos.fromLong(buffer.readLong());
-        steps = ForgeSteps.deserialize(buffer.readInt());
+        steps = new ForgeSteps();
+        steps.deserializeNBT(ByteBufUtils.readTag(buffer));
```

Comment:
Is there a reason this was dropped? It's significantly cheaper to serialize one int than an NBT tag as a string.

Reply:
I've changed since serialization using bit shift didn't work correctly for negative (client deserialization only worked correctly and updated GUI after all three steps weren't null/-1). Since `ForgeSteps` NBT only use 3 integers, i don't think it will cause much overhead, it's *almost* the same as writing them directly to the buffer.

Reply:
No, it's not important, I was mostly curious. If there's a bug then this is case and point for bit shift magic being too complicated for the situation.

(If I've done the math correctly, writing the tag is 33 bytes once you include tag names, header info, and three integers)